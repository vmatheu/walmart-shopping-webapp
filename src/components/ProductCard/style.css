.walmark-product-card {
    margin: 0 0 10px 10px;
}

.walmark-product-card-discount {
    color: white;
    background-color: red;
}

.walmark-product-card-discount-price {
    text-decoration: line-through;
    color: grey;
}
