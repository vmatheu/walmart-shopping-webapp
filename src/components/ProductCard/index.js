import React from 'react';
import { string, number, bool } from 'prop-types';
import './style.css';

const ProductCard =
  ({ id, description, priceHaveDiscount, price, priceWithOutDiscount, brand, image }) => (
    <div className="ard mb-4 shadow-sm walmark-product-card">
      <div className="card-header">
        <h4 className="my-0 font-weight-normal">{description}</h4>
      </div>
      <div className="card-body">
        {priceHaveDiscount ?
          <div className="card-title pricing-card-title">
            <h3> ${price} <span className="walmark-product-card-discount">50%</span></h3>
            <h5 className="walmark-product-card-discount-price"> ${priceWithOutDiscount} </h5>
          </div>
        : <h3 className="card-title pricing-card-title">${price}</h3>}
        <img alt={`imagen del producto ${description}`} src={`http://${image}`} />
        <h6>{brand}</h6>
      </div>
    </div>
);

ProductCard.propTypes = {
  id: number.isRequired,
  description: string.isRequired,
  priceHaveDiscount: bool.isRequired,
  price: number.isRequired,
  priceWithOutDiscount: number.isRequired,
  brand: string.isRequired,
  image: string.isRequired,
};

export default ProductCard;
