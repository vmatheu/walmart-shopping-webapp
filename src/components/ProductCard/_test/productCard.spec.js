import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ProductCard from '..';

const buildProduct =
  (id, brand, description, price, image, priceHaveDiscount, priceWithOutDiscount) =>
  ({ id, brand, description, price, image, priceHaveDiscount, priceWithOutDiscount });

describe('<ProductCard/>', () => {
  describe('should be product card', () => {
    it('should have the same ui structure with one product', () => {
      const product =  buildProduct(1, 'coca cola', 'bebida', 10, 'uriimage', false, 0);
      const wrapper = shallow(<ProductCard {...product} />);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});
