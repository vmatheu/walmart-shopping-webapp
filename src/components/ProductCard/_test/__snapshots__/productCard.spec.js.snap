// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard/> should be product card should have the same ui structure with one product 1`] = `
<div
  className="ard mb-4 shadow-sm walmark-product-card"
>
  <div
    className="card-header"
  >
    <h4
      className="my-0 font-weight-normal"
    >
      bebida
    </h4>
  </div>
  <div
    className="card-body"
  >
    <h3
      className="card-title pricing-card-title"
    >
      $
      10
    </h3>
    <img
      alt="imagen del producto bebida"
      src="http://uriimage"
    />
    <h6>
      coca cola
    </h6>
  </div>
</div>
`;
